lib/object.c

```diff
@@ -188,44 +172,71 @@ tool_rc setup_primary(ESYS_CONTEXT *esys_ctx, ESYS_TR *parent) {
     return rc;
 }
 
+tool_rc tpm2_util_object_fetch_tpk(const char *objectstr,
+        TSSPRIVKEY_OBJ **tpk) {
+    tool_rc rc = tool_rc_success;
+    BIO *input_bio = BIO_new_file(objectstr, "rb");
+    if (!input_bio) {
+        LOG_ERR("Unable to read as BIO file");
+        rc = tool_rc_general_error;
+        goto ret;
+    }
+
+    *tpk = PEM_read_bio_TSSPRIVKEY_OBJ(input_bio, NULL, NULL, NULL);
+    if (*tpk == NULL) {
+        LOG_ERR("Unable to read PEM from provided BIO/file");
+        rc = tool_rc_general_error;
+        goto ret;
+    }
+
```

Comment:
and set rc = tool_rc_success here, and drop the rc = tool_rc_general_error on line 181 and 188.

This way if someone forgets to change the rc in modifications that check for an error that doesn't set rc, they won't accidentally goto ret and return success.